torch/testing/_internal/common_methods_invocations.py

```diff
@@ -13814,6 +13814,9 @@ def error_inputs_mean(op_info, device, **kwargs):
            dtypesIfCUDA=floating_types_and(torch.float16, torch.bfloat16),
            supports_out=False,
            supports_forward_ad=True,
+           # While computing batched gradients, got: vmap: Calling Tensor.as_strided is not supported unless the batch
```

Comment:
@zou3519 @samdow running with functorch also fails with the same error, see `python test/test_ops.py -k test_vmapjvpall_nn_functional_layer_norm`, there seem to be no issues with `python test/test_ops.py -k __rdiv__` however.

The reason this is now failing is this PR makes it more strict as to what is considered a perfect metadata match between the primal and tangent: previously we allowed the strides at a dimension to be different if the size corresponding to that dim is 0 or 1, but now we perform an out-of-place view to make sure the strides are actually the same. So the reason we are running into the error is that we are calling as_strided in a place where we previously were not.